worker/build.rs

```diff
@@ -78,19 +80,26 @@ fn main() {
     #[cfg(target_os = "windows")]
     {
         if !std::path::Path::new("worker/out/msys/bin/make.exe").exists() {
-            let python = if Command::new("where")
-                .arg("python3.exe")
+            let python = if env::var("PYTHON").is_ok() {
+                env::var("PYTHON").unwrap()
+            } else if Command::new("where")
+                .arg("python3")
                 .status()
                 .expect("Failed to start")
                 .success()
             {
-                "python3"
+                "python3".to_string()
             } else {
-                "python"
+                "python".to_string()
             };
 
+            let worker_abs_path = env::current_dir().unwrap();
```

Comment:
```rust
println!("--- mediasoup_out_dir: {}", mediasoup_out_dir);

// => mediasoup_out_dir: /Users/ibc/src/v3-mediasoup/target/debug/build/mediasoup-sys-11923be8cb8ca4cd/out/out
```

That's far from what I want.

Reply:
That is actually precisely what you want

Reply:
I don't see that.

```rust
if !std::path::Path::new("worker/out/msys/bin/make.exe").exists() {
    let python = if let Ok(python) = env::var("PYTHON") {
        python
    } else if Command::new("where")
        .arg("python3")
        .status()
        .expect("Failed to start")
        .success()
    {
        "python3".to_string()
    } else {
        "python".to_string()
    };

    let worker_abs_path = env::current_dir().unwrap();
    let dir = worker_abs_path.join("out/msys");

    if !Command::new(python)
        .arg("scripts\\getmake.py")
        .arg("--dir")
        .arg(dir)
        .status()
        .expect("Failed to start")
        .success()
    {
        panic!("Failed to install MSYS/make")
    }
}

env::set_var(
    "PATH",
    format!(
        "{}\\worker\\out\\msys\\bin;{}",
        env::current_dir()
            .unwrap()
            .into_os_string()
            .into_string()
            .unwrap(),
        env::var("PATH").unwrap()
    ),
);
```

- Here we first check `mediasoup/worker/out/msys/bin/make.exe`.
- If it doesn't exist then we run `getmake.py` and we want to store things in `dir` (let's see below what `dir` should be).
- Then we do this:
  ```rust
  env::set_var(
    "PATH",
    format!(
        "{}\\worker\\out\\msys\\bin;{}",
        env::current_dir()
            .unwrap()
            .into_os_string()
            .into_string()
            .unwrap(),
        env::var("PATH").unwrap()
    ),
  );
```
- There we are **clearly** adding current dir + \\worker\\out\\msys\\bin to the `PATH`.
- So it's very clear that `dir` given to `getmake.py` must point to current dir + \\worker\\out\\msys as well.

Reply:
So everything was wrong before or there is something I miss.

In addition, it looks suspicious to me that `mediasoup_out_dir` in my macOS points to "/Users/ibc/src/v3-mediasoup/target/debug/build/mediasoup-sys-11923be8cb8ca4cd/out/out" (note the double "out" at the end).

Reply:
> Here we first check mediasoup/worker/out/msys/bin/make.exe.

This is fine because it might actually exist there in local development due to being installed by Node.js mediasoup

> If it doesn't exist then we run getmake.py and we want to store things in dir (let's see below what dir should be).

I think this is actually a problem, whoever introduced it should not have downloaded files where they did.

> There we are clearly adding current dir + \worker\out\msys\bin to the PATH.

Current directory is fine, but we need to add the correct location within `OUT_DIR` where we download stuff just like we download PIP, compile static library and do other things in `OUT_DIR`.

> So it's very clear that dir given to getmake.py must point to current dir + \worker\out\msys as well.

It is not clear at all, both old and new code is incorrect as far as I'm concerned. In fact, Rust will likely refuse to publish package at all if it modifies anything outside of `OUT_DIR`, crates are supposed to be self-contained.

Reply:
> In addition, it looks suspicious to me that mediasoup_out_dir in my macOS points to "/Users/ibc/src/v3-mediasoup/target/debug/build/mediasoup-sys-11923be8cb8ca4cd/out/out" (note the double "out" at the end).

Looks like that, yes. One `out` is from Cargo, second is "ours".

Reply:
ok, so I'll **also** add the missing path to `PATH`.

But you know the funniest thing? **I'm trying to fix an issue that NO longer happens in v3 branch**: https://github.com/versatica/mediasoup/pull/1189 (it's basically v3 branch and CI are perfectly passing).

Reply:
It might, but probably because CI worked before too due to GitHub Actions containing correct `make`, but when people install mediasoup on their machines, suddenly things break because `make` in `PATH` is nmake or whatever the thing is that is on Windows by default.

Reply:
> Looks like that, yes. One out is from Cargo, second is "ours".

But how can it be wrong? I mean, `mediasoup_out_dir` is being used by build.rs to give env `MEDIASOUP_OUT_DIR` to `make libmediasoup-worker`. How is possible that it was wrong all this time? I mean, can I just make `mediasoup_out_dir` be `out_dir`:

```diff
- let mediasoup_out_dir = format!("{}/out", out_dir.replace('\\', "/"));
+ let mediasoup_out_dir = out_dir.replace('\\', "/");
```

and things will work?

Reply:
That part isn't wrong, I meant `get_make` and where it places downloaded files wasn't correct.

Reply:
This is gonna be fine: https://github.com/versatica/mediasoup/pull/1187/commits/cca0dabc2e00f12f94eb767c853a9612b6843cde

Reply:
Rust is amazing. I declare a `let dir` and it later complains because I have "moved" it so I cannot use it in two places and must clone it in one of them.

```
error[E0382]: borrow of moved value: `dir`
   --> worker\build.rs:109:39
    |
94  |             let dir = format!("{}/msys", mediasoup_out_dir.replace('\\', "/"));
    |                 --- move occurs because `dir` has type `std::string::String`, which does not implement the `Copy` trait
...
99  |                 .arg(dir)
    |                      --- value moved here
...
109 |                 format!("{}\\bin;{}", dir, env::var("PATH").unwrap()),
    |                                       ^^^ value borrowed here after move
    |
    = note: this error originates in the macro `$crate::__export::format_args` which comes from the expansion of the macro `format` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider cloning the value if the performance cost is acceptable
    |
99  |                 .arg(dir.clone())
    |                         ++++++++
```